stdlib/Profile/src/Profile.jl

```diff
@@ -560,26 +560,33 @@ function tree!(root::StackFrameTree{T}, all::Vector{UInt64}, lidict::Union{LineI
     return root
 end
 
-# Print a "branch" starting at a particular level. This gets called recursively.
+# Print the stack frame tree starting at a particular root. Uses a worklist to
+# avoid stack overflows.
 function tree(io::IO, bt::StackFrameTree, level::Int, cols::Int, fmt::ProfileFormat, noisefloor::Int)
-    level > fmt.maxdepth && return
-    isempty(bt.down) && return
-    # Order the line information
-    nexts = collect(values(bt.down))
-    lilist = collect(frame.frame for frame in nexts)
-    counts = collect(frame.count for frame in nexts)
-    # Generate the string for each line
-    strs = tree_format(lilist, counts, level, cols)
-    # Recurse to the next level
-    for i in liperm(lilist)
-        down = nexts[i]
-        count = down.count
-        count < fmt.mincount && continue
-        count < noisefloor && continue
-        str = strs[i]
-        println(io, isempty(str) ? "$count unknown stackframe" : str)
-        noisefloor_down = fmt.noisefloor > 0 ? floor(Int, fmt.noisefloor * sqrt(count)) : 0
-        tree(io, down, level + 1, cols, fmt, noisefloor_down)
+    worklist = Tuple{StackFrameTree, Int, Int, Union{String, Nothing}}[
```

Comment:
```suggestion
    worklist = [(bt, level, noisefloor, "")]
```

No sense in reducing performance and introducing pointless complexity.

But I don't think this translation is correct: it doesn't seem expected that `str` would suddenly get appended to the worklist, when it wasn't there before the translation. It looks like this would print all level 0 items, followed by all level 1 items, followed by all level 2 items, and so on. (indeed, running this PR on `@profile peakflops()`, that's exactly the behavior that I see happen)